Merging in updateToEmberThreeFour

rupdegraff opened the pull request:
Updated to Ember 3.4.1 and added several new features, including: 

- ability to prevent specific routes from being added to `sitemap.xml`
- adding a custom change frequency value
- choosing whether to display the sitemap-autogenerator log
- adding custom priorities values to specific routes
- setting your own default priority value for the generator
- updating the `README` file to detail these new features

The bulk of these changes are in `index.js`, while ember-cli-update did the heavy lifting on bringing sitemap-autogenerator to Ember 3.

blkeller commented on README.md:
```diff
@@ -19,6 +20,33 @@ To install simply run:
+      'contact': 'contact',
+      'algorithmictradedeveloper': 'algorithmictradedeveloper',
+      'careers': 'careers'
+    },
```
Thanks for working on this!

In the course of testing this PR, I'm trying to follow your new instructions for using `ignore`.  The way I'm thinking about this, the end goal is to prevent certain paths from showing up in the 'sitemap.xml'.  Given that, I was expecting to provide an array of paths, and I'm not sure I understand what the keys and values are supposed to represent in the `ignore` object.  Can you please clarify this in the documentation?

rupdegraff commented:
I added provisions for someone not having added the 'sitemap-autogenerator' config object to the ENV object in environment.js.  A warning message will appear during the npm run build process stating that sitemap-autogenerator is installed but not properly configured.

The message states that a sitemap.xml will be created without any custom settings and encourages developers to visit the documentation regarding adding sitemap-autogenerator to the project ENV.

rupdegraff commented:
@blkeller Thanks for testing `sitemap-autogenerator` and finding the issues you listed.  I believe I've addressed each of the in subsequent commits and have also rewritten the primary parsing algorithm, greatly reducing the code length and removing restrictions governing nested route depth.

Would you please retest the addon at your earliest convenience.  Thanks!
